polaris-core/src/main/java/org/apache/polaris/core/auth/AllowAllAuthorizer.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.core.auth;
+
+import jakarta.annotation.Nonnull;
+import java.util.List;
+import org.apache.polaris.core.PolarisConfigurationStore;
+import org.apache.polaris.core.persistence.resolver.PolarisResolutionManifest;
+
+public class AllowAllAuthorizer implements PolarisAuthorizer {
```

Comment:
Since we have 2 authz implementations now, shouldn't we annotate `PolarisAuthorizer` with `@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, property = "class")`?

Reply:
Unfortunately, the original `PolarisAuthorizer` impl. takes a constructor arguments. I do not this DW is able to handle that. With CDI (e.g. HK2) we should be able to do that, but for now I think we have to use a no-arg factory injected into the config and then create an authorizer instances from the factory. 

This will add some boiler-plate code, which is why I did the manual constructor invocation in the first revision of this PR... but I'm fine updating to more DW-native injection patterns.

Reply:
Okay, whatever works best for you.

Reply:
Added a factory with `@JsonTypeInfo`. It turned out extending `Discoverable` was not required with `JsonTypeInfo.Id.CLASS` :)